### PR TITLE
[DO NOT MERGE]: update to MSAL 3.x

### PR DESCRIPTION
**Pull request by jmprieur**
## Purpose
Update the sample to MSAL 3.x

### CODE REVIEW

File: iwa-console/AuthenticationConfig.cs [hunk 1]
```diff
@@ -35,41 +36,12 @@ namespace iwa_console
     /// </summary>
     public class AuthenticationConfig
     {
-        /// <summary>
```
Comment (jmprieur) [inline iwa-console/AuthenticationConfig.cs]: The Authentication Config is now simplified as we use the PublicClientApplicationOptions. It now only contains the code to read the config from the Json file.

File: iwa-console/packages.config [hunk 2]
```diff
@@ -9,7 +9,7 @@
   <package id="Microsoft.Extensions.FileProviders.Physical" version="2.1.1" targetFramework="net461" />
   <package id="Microsoft.Extensions.FileSystemGlobbing" version="2.1.1" targetFramework="net461" />
   <package id="Microsoft.Extensions.Primitives" version="2.1.1" targetFramework="net461" />
-  <package id="Microsoft.Identity.Client" version="2.6.2" targetFramework="net461" />
+  <package id="Microsoft.Identity.Client" version="1.0.0-localbuild" targetFramework="net461" />
```
Comment (jmprieur) [inline iwa-console/packages.config]: The NuGet package was generated locally (and I don't know how to change its number to 3.0.0-preview

File: iwa-console/PublicAppUsingIntegratedWindowsAuthentication.cs [hunk 3]
```diff
@@ -43,11 +43,11 @@ public class PublicAppUsingIntegratedWindowsAuthentication
         /// <remarks>
         /// For more information see https://aka.ms/msal-net-iwa
         /// </remarks>
-        public PublicAppUsingIntegratedWindowsAuthentication(PublicClientApplication app)
+        public PublicAppUsingIntegratedWindowsAuthentication(IPublicClientApplication app)
         {
             App = app;
         }
-        protected PublicClientApplication App { get; private set; }
+        protected IPublicClientApplication App { get; private set; }
```
Comment (jmprieur) [inline iwa-console/PublicAppUsingIntegratedWindowsAuthentication.cs]: This is the big (breaking) change: Now that the build returns an IPublicClientApplication, everywhere we had PublicClientApplication in the code, we change it to IPublicClientApplication

File: iwa-console/packages.config (hunk 2 above)
Comment (bgavrilMS) [reply iwa-console/packages.config]: When you build and pack, add /p:MsalClientSemVer=3.0.0-preview to the msbuild cmd line

File: iwa-console/PublicAppUsingIntegratedWindowsAuthentication.cs (hunk 3 above)
Comment (bgavrilMS) [reply iwa-console/PublicAppUsingIntegratedWindowsAuthentication.cs]: I agree it's a breaking change, but why is it big? Just add an "I" :).

Comment (jmprieur) [reply iwa-console/PublicAppUsingIntegratedWindowsAuthentication.cs]: this takes time :)

**Pull request merged**